ansible/templates/usb/scripts/computenode.sh.j2

```diff
@@ -122,61 +142,128 @@ function install_archives() {
 }
 
 function __zk() {
-	zkcli="${USB_COPY}/scripts/zookeepercli"
-	args="${@}"
-
-	"${zkcli}" -servers "${CFGDB_IP}:2181" -auth_usr "esdc" -auth_pwd "${ESDC_INSTALL_PASSWORD}" ${args}
+	"${QUERY_CFGDB}" "${@}"
 }
 
 function _zk() {
-	if ! __zk -c "${@}" 2> /dev/null; then
+	if ! __zk "${@}" 2> /dev/null; then
 		fatal "Error: cfgdb command \"${@}\" failed"
 	fi
 }
 
+function _zk_noerror() {
+	if ! __zk "${@}" 2> /dev/null; then
+		# return success anyway
+		return 0
+	fi
+}
+
+# omit SSL certificate verification
+function _zk_noverify() {
+	_zk_noerror -k "${@}"
+}
+
+test_cfgdb_connect() {
+	if [[ -z "$(_zk_noverify get /esdc/settings)" ]]; then
+		fatal "Unable to query cfgdb"
+	fi
+}
+
+test_port_open() {
+	local host="${1}"
+	local port="${2}"
+	local waittime=5
+
+	"${TIMEOUT}" "${waittime}" nc -z "${host}" "${port}"
+}
+
+function setup_svc_ssl_cert() {
+	# download SSL cert
+	local svc_cert="$(_zk_noverify get /esdc/settings/security/services_ssl_cert)"
+
+	# the only expected case for not getting the certificate
+	# is adding v3.0.0 or later compute node to pre-v3.0.0 management
+	if [[ -z "${svc_cert}" ]]; then
+		# node must have SSL enabled...
+		# fail loudly then
+		fatal "No certificate retrieved from configuration database. Installation cannot continue."
+	fi
+
+	# save the public cert into openssl trusted certs
+	echo "${svc_cert}" > "${SVC_CERT_FILE}"
+	# rehash certificate
+	ln -s "${SVC_CERT_FILE}" "${OPENSSL_HOME}/certs/${SVC_CERT_FILE_NAME}"
+	ln -s "${SVC_CERT_FILE_NAME}" "${OPENSSL_HOME}/certs/$("${OPENSSL}" x509 -hash -noout -in "${SVC_CERT_FILE}").0"
```

Comment:
This won't yield expected result in the current working directory.

Reply:
I believe it will. Try it.

Reply:
OK. I've never used it this way